File: clients/python/examples/collections-replace-one-upsert.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Replace a document
result = collection.replace_one(
    {
        "$and": [
            {"is_checked_out": False},
            {"number_of_pages": {"$lt": 300}},
        ]
    },
    {"is_checked_out": True, "borrower": "Brook Reed"},
    upsert=True,
)

print(result)

File: clients/csharp/examples/collections-replace-one-upsert.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Replace a document
    var filterBuilder = Builders<Document>.CollectionFilter;
    var filter = filterBuilder.And(
      filterBuilder.Eq("is_checked_out", false),
      filterBuilder.Lt("number_of_pages", 300)
    );
    var replacement = new Document()
    {
      { "is_checked_out", true },
      { "borrower", "Brook Reed" },
    };
    var replaceOptions = new CollectionReplaceOneOptions<Document>()
    {
      Upsert = true,
    };
    var result = await collection.ReplaceOneAsync(
      filter,
      replacement,
      replaceOptions
    );

    Console.WriteLine(result.MatchedCount);
    Console.WriteLine(result.ModifiedCount);
    Console.WriteLine(result.UpsertedId);
  }
}
